tmail-backend/saas/saas-api/src/test/java/com/linagora/tmail/saas/mailet/LocalDateFormatterTest.java

```diff
@@ -0,0 +1,44 @@
+/********************************************************************
+ *  As a subpart of Twake Mail, this file is edited by Linagora.    *
+ *                                                                  *
+ *  https://twake-mail.com/                                         *
+ *  https://linagora.com                                            *
+ *                                                                  *
+ *  This file is subject to The Affero Gnu Public License           *
+ *  version 3.                                                      *
+ *                                                                  *
+ *  https://www.gnu.org/licenses/agpl-3.0.en.html                   *
+ *                                                                  *
+ *  This program is distributed in the hope that it will be         *
+ *  useful, but WITHOUT ANY WARRANTY; without even the implied      *
+ *  warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR         *
+ *  PURPOSE. See the GNU Affero General Public License for          *
+ *  more details.                                                   *
+ *******************************************************************/
+
+package com.linagora.tmail.saas.mailet;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.ZonedDateTime;
+import java.util.Locale;
+
+import org.junit.jupiter.api.Test;
+
+public class LocalDateFormatterTest {
+    @Test
+    void englishLocalDateFormatTest() {
+        ZonedDateTime zonedDateTime = ZonedDateTime.parse("2015-10-30T14:12:00Z");
+
+        assertThat(zonedDateTime.format(I18NDSNBounce.localizedDateFormatter(Locale.ENGLISH)))
+            .isEqualTo("Friday, 30 October 2015 14:12:00 Z");
+    }
+
+    @Test
+    void frenchLocalDateFormatTest() {
+        ZonedDateTime zonedDateTime = ZonedDateTime.parse("2015-10-30T14:12:00Z");
+
+        assertThat(zonedDateTime.format(I18NDSNBounce.localizedDateFormatter(Locale.FRENCH)))
+            .isEqualTo("vendredi, 30 octobre 2015 14:12:00 Z");
```

Comment:
I tried to convert the time `14:12:00 Z` to user's local time. But Locale is not sufficient enough to infer TimeZone... Maybe we can live with the UTC time for now, until we have some information like user's timezone?

Reply:
We can reuse the timezone used in the Date header IMO